akka-stream/src/main/scala/akka/stream/impl/FanIn.scala

```diff
@@ -151,6 +155,8 @@ private[akka] object FanIn {
 
     def isCancelled(input: Int): Boolean = cancelled(input)
 
+    def isAllCompleted(): Boolean = (0 to inputCount - 1).map(completed).forall(_ == true)
```

Comment:
No I just meant that even though this class will be deprecated, the test cases reproducing the original issue will stay and will be very useful.